navigator_engine/common/graph_loader.py

```diff
@@ -321,6 +336,17 @@ def validate_pluggable_logic(node, network):
 
         assert arg_action_is_child, f"{node.ref} has bad function {function_string}"
 
+    elif function_name.startswith('check_not_skipped'):
+        ...  # We already validate args in _create_check_skips_action()
+
+    elif function_name.startswith('check_resource_key'):
+        resource_types = function_args[0]
+        if type(resource_types) is str:
+            resource_types = [resource_types]
+        for resource_type in resource_types:
+            assert resource_type in VALID_RESOURCE_TYPES, \

```

Comment:
Is assert statement the correct way to validate program input? Using an if-statement and raising ValueError on invalid input would be more informative.

Reply:
It is suggested by Python docs that asserts would not be used in validating input: https://wiki.python.org/moin/UsingAssertionsEffectively The reasons for this suggestion are not relevant for navigator_engine at the moment and the practice is unified in the code base. Resolving conversation.